Sources/ConsoleKit/Utilities/ConsoleLogger.swift

```diff
@@ -116,12 +178,29 @@ extension LoggingSystem {
     ///
     /// - Parameters:
     ///   - console: The console the logger will log the messages to.
-    ///   - level: The minimum level of message that the logger will output. This defaults to `.debug`, the lowest level.
-    ///   - metadata: Extra metadata to log with the message. This defaults to an empty dictionary.
-    public static func bootstrap(console: Console, level: Logger.Level = .info, metadata: Logger.Metadata = [:]) {
-        self.bootstrap { label in
-            return ConsoleLogger(label: label, console: console, level: level, metadata: metadata)
-        }
+    ///   - level: The minimum level of message that the logger will output. This defaults to `.info`.
+    ///   - metadata: Extra metadata to log with all messages. This defaults to an empty dictionary.
+    ///   - metadataProvider: The metadata provider to bootstrap the logging system with.
+    public static func bootstrap(console: Console, level: Logger.Level = .info, metadata: Logger.Metadata = [:], metadataProvider: Logger.MetadataProvider? = nil) {
+        self.bootstrap({ (label, metadataProvider) in
+            return ConsoleLogger(label: label, console: console, level: level, metadata: metadata, metadataProvider: metadataProvider)
+        }, metadataProvider: metadataProvider)
+    }
```

Comment:
Nope, this time you _did_ break public API, there's no `Logger`-provided extension that covers the non-`metadataProvider` version for this one. You have to provide the original version instead of defaulting:
```suggestion
    ///   - level: The minimum level of message that the logger will output. This defaults to `.info`.
    ///   - metadata: Extra metadata to log with all messages. This defaults to an empty dictionary.
    public static func bootstrap(console: Console, level: Logger.Level = .info, metadata: Logger.Metadata = [:]) {
        self.bootstrap(console: console, level: level, metadata: metadata, metadataProvider: nil)
    }
    
    /// Bootstraps a `ConsoleLogger` to the `LoggingSystem`, so that logger will be used in `Logger.init(label:)`.
    ///
    ///     LoggingSystem.boostrap(console: console)
    ///
    /// - Parameters:
    ///   - console: The console the logger will log the messages to.
    ///   - level: The minimum level of message that the logger will output. This defaults to `.info`.
    ///   - metadata: Extra metadata to log with all messages. This defaults to an empty dictionary.
    ///   - metadataProvider: The metadata provider to bootstrap the logging system with.
    public static func bootstrap(console: Console, level: Logger.Level = .info, metadata: Logger.Metadata = [:], metadataProvider: Logger.MetadataProvider?) {
        self.bootstrap({ (label, metadataProvider) in
            ConsoleLogger(label: label, console: console, level: level, metadata: metadata, metadataProvider: metadataProvider)
        }, metadataProvider: metadataProvider)
    }
```

Reply:
Is adding an argument with a default value not backwards compatible under some circumstances?

Reply:
Dear Gods, here we go again...! 😂 Fortunately, @MahdiBM gave a quite detailed explanation in his pitch about fixing the problem in the language: https://forums.swift.org/t/pitch-adding-new-parameters-w-default-values-should-not-break-api/66847